### ISSUE TITLE
TDO file conversion

### ISSUE THREAD

**Issue by jskeet** [jskeet/DemoCode#103]
From #100:

> The drum kit files I've gotten from some of the aftermarket suppliers (Vexpressions, Michael Shack etc) are all in a .tdo format, but V-Drum Explorer saves/opens .vdrum files.
> Is there a way for V-Drum explorer to access the .tdo files directly so I can mix and match kits from different sources?
> If not, is that in the road map at all?

**Comment by jskeet** [comment]
Currently there's no conversion between .tdo files and .vdrum files. I haven't looked into the .tdo format, although I wouldn't be at all surprised if it were just a memory dump of the kit.
I'll look into it further.

One thing I *don't* want to risk is V-Drum Explorer being used to violate the copyright of aftermarket suppliers. Not having used these myself, am I right in saying that folks who wanted to would already be able to just email .tdo files to each other anyway, and that converting to .vdrum wouldn't actually make this any easier?

**Comment by rebelsoxuk** [comment]
Yes, I believe the .tdo format is just a memory dump (and apparently now includes the sample mapping as well because pretty much all of them are including samples for TD-17/25/50's)
This guy makes kits and gives them away if you needed some .tdo's to work with for the different modules:
https://simonayton.com/roland-v-drums-free-kit-downloads

Yeah, people can just share them via email if they wanted to. So converting to .vdrums not gonna change anything for the pirates :( Luckily that doesn't appear to be a major issue in the FB groups that I've seen :)

I came across a Yahoo group years ago that apparently was being used to do exactly that... I reported it to the Vexpressions team but they seemed powerless to get it pulled down by Yahoo :(

Regards

**Comment by jskeet** [comment]
That's fascinating in terms of the samples. Those wouldn't be representable in .vkit files, as I don't have access to samples (either loading or saving) as far as I'm aware. (If anyone *does* know a way of loading or saving samples over MIDI, I'd love to hear!)
But everything other than the samples would be available, at least.

**Comment by rebelsoxuk** [comment]
No, no one is loading/saving samples over MIDI. They're just being referenced/mapped from the Import folder that needs be put on the SD card.

**Comment by jskeet** [comment]
Right. I need to have a think about whether it would make sense to try to include this within the vkit format, or have a whole separate format for td0.

I think the first step will be to try to load a td0 as a vkit while skipping the samples though.

**Comment by rebelsoxuk** [comment]
If you want/need any testing done from my end, just let me know :)
